fuzzyfinder.go

```diff
@@ -614,9 +630,14 @@ func (f *finder) find(slice interface{}, itemFunc func(i int) string, opts []Opt
 	defer cancel()
 
 	inited := make(chan struct{})
-	if opt.hotReload && rv.Kind() == reflect.Ptr {
-		rvv := reflect.Indirect(rv)
-		items, matched = makeItems(rvv.Len())
+	if opt.hotReload {
+		if rv.Kind() == reflect.Ptr {
```

Comment:
I feel that it's able to put an abstraction for making items.
For example, 

``` go
type itemReloader interface {
	reload() ([]string, []matching.Matched)
}

type sliceItemReloader struct {
	itemFunc func(int) string
	rv       reflect.Value
}

func (r *sliceItemReloader) reload() ([]string, []matching.Matched) {
	sliceLen := reflect.Indirect(r.rv).Len()

	items := make([]string, sliceLen)
	matched := make([]matching.Matched, sliceLen)
	for i := 0; i < sliceLen; i++ {
		items[i] = r.itemFunc(i)
		matched[i] = matching.Matched{Idx: i} //nolint:exhaustivestruct
	}

	return items, matched
}

func newSliceItemReloader(itemFunc func(int) string, rv reflect.Value) itemReloader {
	return &sliceItemReloader{itemFunc: itemFunc, rv: rv}
}

type funcItemReloader struct {
	itemFunc func() []string
}

func (r *funcItemReloader) reload() ([]string, []matching.Matched) {
	items := r.itemFunc()
	matched := make([]matching.Matched, len(items))
	for i := 0; i < len(items); i++ {
		matched[i] = matching.Matched{Idx: i} //nolint:exhaustivestruct
	}

	return items, matched
}

func newFuncItemReloader(rv reflect.Value) itemReloader {
	return &funcItemReloader{itemFunc: rv.Interface().(func() []string)}
}
```

Reply:
again, i am not sure i see the value of having such an abstraction in the lib. So much easier to just leave it to the caller.